api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -2969,4 +2969,178 @@ public void saveOrder_shouldPassIfAnKnownDrugOrderForTheSameDrugFormulationSpeci
+		Concept nonCodedDrugConcept = orderService.getNonCodedDrugConcept();
+		assertNotNull(nonCodedDrugConcept);
+		assertThat(nonCodedDrugConcept.getConceptId(), is(5584));
+		assertEquals(nonCodedDrugConcept.getName().getName(), "DRUG OTHER");
+		
+	}
+	
+	/**
+	 * @verifies pass if an active drug order for the same coded concept, care setting and different drug non coded exists
+	 * @see OrderService#saveOrder(Order, OrderContext)
+	 */
+	@Test
+	public void saveOrder_shouldPassForANewNonCodedDrugOrderIfAnotherActiveNonCodedDrugOrderAlreadyExists()
+	        throws Exception {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-nonCodedDrugs.xml");
+		final Patient patient = patientService.getPatient(2);
+		final Concept nonCodedConcept = conceptService.getConcept(5584);
+		//sanity check that we have an active order for the same concept
+		DrugOrder duplicateOrder = (DrugOrder) orderService.getOrder(584);
+		assertTrue(duplicateOrder.isActive());
+		assertEquals(nonCodedConcept, duplicateOrder.getConcept());
+		
+		DrugOrder drugOrder = new DrugOrder();
+		drugOrder.setPatient(patient);
+		drugOrder.setCareSetting(orderService.getCareSetting(1));
+		drugOrder.setConcept(nonCodedConcept);
+		drugOrder.setDrugNonCoded("non coded drug paracetemol");
+		drugOrder.setEncounter(encounterService.getEncounter(6));
+		drugOrder.setOrderer(providerService.getProvider(1));
+		drugOrder.setCareSetting(duplicateOrder.getCareSetting());
+		drugOrder.setDrug(duplicateOrder.getDrug());
+		drugOrder.setDose(duplicateOrder.getDose());
+		drugOrder.setDoseUnits(duplicateOrder.getDoseUnits());
+		drugOrder.setRoute(duplicateOrder.getRoute());
+		drugOrder.setFrequency(duplicateOrder.getFrequency());
+		drugOrder.setQuantity(duplicateOrder.getQuantity());
+		drugOrder.setQuantityUnits(duplicateOrder.getQuantityUnits());
+		drugOrder.setNumRefills(duplicateOrder.getNumRefills());
+
+		Order savedOrder = orderService.saveOrder(drugOrder, null);
+
+		assertNotNull(orderService.getOrder(savedOrder.getOrderId()));
+	}
+
+	/**
+	 * @verifies fail if an active drug order for the same coded concept, care setting and drug non coded exists
+	 * @see OrderService#saveOrder(Order, OrderContext)
+	 */
+	@Test
+	public void saveOrder_shouldFailIfAnActiveDrugOrderForTheSameConceptAndDrugNonCodedAndCareSettingExists()
+			throws Exception {
+		executeDataSet("org/openmrs/api/include/OrderServiceTest-nonCodedDrugs.xml");
+		final Patient patient = patientService.getPatient(2);
+		final Concept nonCodedConcept = conceptService.getConcept(5584);
+		//sanity check that we have an active order for the same concept
+		DrugOrder duplicateOrder = (DrugOrder) orderService.getOrder(584);
+		assertTrue(duplicateOrder.isActive());
+		assertEquals(nonCodedConcept, duplicateOrder.getConcept());
+
+		DrugOrder drugOrder = new DrugOrder();
+		drugOrder.setPatient(patient);
+		drugOrder.setCareSetting(orderService.getCareSetting(1));
```

Comment:
You should be copying care setting and other fields values from the existing to guarantee they are always the same